Rubberduck.Core/UI/UnitTesting/TestExplorerViewModel.cs

```diff
@@ -132,6 +139,43 @@ public TestExplorerGrouping TestGrouping
             }
         }
 
+        //public System.Collections.ObjectModel.ObservableCollection<System.Windows.Media.Imaging.BitmapImage> OutcomeFilters { get; }
+        public System.Collections.ObjectModel.ObservableCollection<object> OutcomeFilters { get; }
+
+        private TestOutcome? _filtering = null;
+
+        public TestOutcome? TestFiltering
+        {
+            get => _filtering;
+            set
+            {
+                if (value == _filtering)
+                {
+                    return;
+                }
+
+                _filtering = value;
+                Tests.Refresh();
+                OnPropertyChanged();
+            }
+        }
+
+        private string _testNameFilter = string.Empty;
+        public string TestNameFilter
+        {
+            get => _testNameFilter;
+            set
+            {
+                if (_testNameFilter != value)
+                {
+                    _testNameFilter = value;
+                    OnPropertyChanged();
+                    Tests.Filter = FilterResults;
+                    OnPropertyChanged(nameof(Tests));
```

Comment:
`OnPropertyChanged` is not the correct tool to use here. Tests is an ICollectionView. That interface exposes `.Refresh`, which is semantically more appropriate (and also used in the `TestFiltering` setter).

Reply:
Noted. [Reviewed the documentation](https://docs.microsoft.com/en-us/dotnet/api/system.componentmodel.icollectionview.refresh?view=netframework-4.7#System_ComponentModel_ICollectionView_Refresh) for it. I'm assuming internally `Refresh` takes care of `OnPropertyChanged`.